resolver/cel.go

```diff
@@ -146,16 +147,64 @@ func (plugin *CELPlugin) LibraryName() string {
 	return plugin.Name
 }
 
-func (f *CELFunction) CELArgs() []*cel.Type {
-	ret := make([]*cel.Type, 0, len(f.Args))
-	for _, arg := range f.Args {
-		ret = append(ret, ToCELType(arg))
+type CELPluginFunctionSignature struct {
+	ID     string
+	Args   []*cel.Type
+	Return *cel.Type
+}
+
+// Signatures returns signature for overload function.
+// If there is an enum type, we need to prepare functions for both `opaque<int>` and `int` patterns.
+func (f *CELFunction) Signatures() []*CELPluginFunctionSignature {
+	sigs := f.signatures(f.Args, f.Return)
+	sigMap := make(map[string]struct{})
+	ret := make([]*CELPluginFunctionSignature, 0, len(sigs))
+	for _, sig := range sigs {
+		if _, exists := sigMap[sig.ID]; exists {
+			continue
+		}
+		ret = append(ret, sig)
+		sigMap[sig.ID] = struct{}{}
 	}
 	return ret
 }
 
-func (f *CELFunction) CELReturn() *cel.Type {
-	return ToCELType(f.Return)
+func (f *CELFunction) signatures(args []*Type, ret *Type) []*CELPluginFunctionSignature {
+	var sigs []*CELPluginFunctionSignature
+	for idx, arg := range args {
+		if arg.Kind == types.Enum {
+			sigs = append(sigs,
+				f.signatures(
+					append(append(append([]*Type{}, args[:idx]...), Int32Type), args[idx+1:]...),
```

Comment:
super-nit, `append(append(args[:idx], Int32Type), args[idx+1:]...)` could be more readable